.github/workflows/main.yml

```diff
@@ -35,65 +35,53 @@ jobs:
         ports:
           - 6379:6379
     steps:
-    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-    - uses: actions/checkout@v2
-
-    # Should not be needed anymore when https://github.com/NLESC-JCER/cpp2wasm/issues/1 is fixed
-    - name: Generate source code
-      uses: docker://nlesc/pandoc-tangle
-      with:
-        args: --preserve-tabs README.md INSTALL.md
-
-    - uses: actions/setup-python@v1
-      with:
-        python-version: '3.x' # Version range or exact version of a Python version to use, using SemVer's version range syntax
-        architecture: 'x64' # optional x64 or x86. Defaults to x64 if not specified
-
-    - name: Install Python dependencies
-      run: make py-deps && pip install httpie
-
-    - name: Run Python example
-      run: make test-py
-
-    - name: Start web application in background
-      run: make run-webapp &
-
-    - name: Test web application
-      run: http --ignore-stdin -f localhost:5001 epsilon=0.001 guess=-20
-
-    - name: Start web service in background
-      run: make run-webservice &
-
-    - name: Test web service
-      run: make test-webservice
-
-    - name: Start Celery web app in background
-      run: make run-celery-webapp &
-
-    - name: Start Celery worker in background
-      run: |
-        cd src/py
-        PYTHONPATH=$PWD/../.. celery -A tasks worker &
-        cd ../..
-
-    - name: Test Celery web app
-      run: |
-        http --ignore-stdin -hf localhost:5000 epsilon=0.001 guess=-20 | tee response.txt
-        # Parse result url from response
-        RESULT_URL=$(cat response.txt |grep Location |awk '{print $2}')
-        sleep 2
-        http --ignore-stdin $RESULT_URL
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v1
+        with:
+          python-version: "3.x" # Version range or exact version of a Python version to use, using SemVer's version range syntax
+          architecture: "x64" # optional x64 or x86. Defaults to x64 if not specified
+
+      - name: Install Python dependencies
+        run: make py-deps && pip install httpie
+
+      - name: Run Python example
+        run: make test-py
+
+      - name: Start web application in background
+        run: make run-webapp &
```

Comment:
It is better to keep some log for these background processes and save them as artifact.

`make run-webapp > ./artifacts/run-webapp.log 2>&1 &`

Reply:
Tried to add, but log is not uploaded on failure